src/main/java/uk/gov/hmcts/ccd/domain/model/definition/FieldType.java

```diff
@@ -129,6 +134,34 @@ public void setCollectionFieldType(FieldType collectionFieldType) {
         this.collectionFieldType = collectionFieldType;
     }
 
+    public Optional<CommonField> getNestedField(String path) {
+        if (StringUtils.isBlank(path) || this.getChildren().isEmpty()) {
+            return Optional.empty();
+        }
+        List<String> pathElements = Arrays.stream(path.trim().split("\\.")).collect(toList());
+
+        return reduce(this.getChildren(), pathElements);
+    }
+
+    private Optional<CommonField> reduce(List<CaseField> caseFields, List<String> pathElements) {
+        String firstPathElement = pathElements.get(0);
+        Optional<CaseField> optionalCaseField = caseFields.stream().filter(e -> e.getId().equals(firstPathElement)).findFirst();
+        if (optionalCaseField.isPresent()) {
+            CommonField caseField = optionalCaseField.get();
+
+            if (pathElements.size() == 1) {
+                return Optional.of(caseField);
+            } else {
+                List<CaseField> newCaseFields = caseField.getFieldType().getChildren();
+                List<String> tail = pathElements.subList(1, pathElements.size());
+
+                return reduce(newCaseFields, tail);
+            }
```

Comment:
instead of subList() better to use index like below then it will give some performance benefit as well.
reduce(List<CaseField> caseFields, List<String> pathElements, int startIndex)
........
return reduce(newCaseFields, pathElements, index + 1);

Reply:
Won't action this as part of this ticket as is part of an existing implementation that was copied over from `CommonField`. In latest commit I've updated `CommonField` to now use this one as well so it's not all duplicated.